controllers/events.js

```diff
@@ -0,0 +1,207 @@
+  try {
+    const eventData = await apiService.post("/rooms", payload);
+    const propertiesToRemove = ["customer_id", "app_id", "recording_info", "template_id", "template", "customer"];
+    const event = removeUnwantedProperties(propertiesToRemove, eventData);
+    const eventDataFromDB = await eventQuery.createEvent({ room_id: eventData.id, created_by: userId, ...event });
+    return res.status(201).json(eventDataFromDB);
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't create event. Please try again later",
+    });
+  }
+};
+
+/**
+ * Retrieves all events that match the query parameters provided in the HTTP request query string.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The events data in JSON format.
+ * @throws {Error} If an error occurs while retrieving the events data.
+ */
+const getAllEvents = async (req, res) => {
+  /**
+   * @type {boolean} - enabled: determines whether the rooms should be enabled or disabled.
+   * @type {number} - limit: The maximum number of rooms to retrieve.
+   * @type {string} - offset: The starting point for retrieving rooms.
+   */
+  const { enabled, limit, offset } = req.query;
+  try {
+    const start = offset || "";
+    const limitOfRooms = limit || GET_ALL_EVENTS_LIMIT_MIN;
+    const isEnabled = enabled || false;
+    const eventsData = await apiService.get(`/rooms?limit=${limitOfRooms}&enabled=${isEnabled}&start=${start}`);
+    if (eventsData.data) {
+      const propertiesToRemove = ["customer_id", "app_id", "recording_info", "template_id", "template", "customer"];
+      const events = removeUnwantedProperties(propertiesToRemove, eventsData.data);
+
+      const filteredEventsData = {
+        limit: eventsData.limit,
+        last: eventsData.last,
+        data: events.map(({ id, ...event }) => ({
+          id,
+          room_id: id,
+          ...event,
+        })),
+      };
+      return res.status(200).json(filteredEventsData);
+    }
+    return res.status(200).json(eventsData);
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't get events. Please try again later",
+    });
+  }
+};
+
+/**
+ * Generates a token for the specified event and user information.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} An object containing the generated token and a success message in JSON format.
+ * @throws {Error} If an error occurs while generating the token.
+ */
+const joinEvent = async (req, res) => {
+  const { roomId, userId, role } = req.body;
+  const payload = { room_id: roomId, user_id: userId, role };
+  try {
+    const token = tokenService.getAuthToken(payload);
+    res.status(201).json({
```

Comment:
what if the status code is anything apart from 201?  Lets say something like 200 where you don't receive back any data ? what then ?

Reply:
We will always receive the token. Let me know if I am missing something out here.